Assets/Scripts/Core/GridCombatSystem.cs

```diff
@@ -149,17 +149,22 @@ private enum State {
 				}
 			}
 
+			foreach(Tilemap.Node node in grid.GetAllGridObjects()) {
+				if(node.walkable) {
+					gridPathfinding.SetWalkable(node.gridX, node.gridY, true);
+				} else {
+					gridPathfinding.SetWalkable(node.gridX, node.gridY, false);
+				}
+			}
+
 			int maxMoveDistance = unitGridCombat_.GetActionPoints() + 1;
 			for(int x = unitX - maxMoveDistance; x <= unitX + maxMoveDistance; x++) {
 				for(int y = unitY - maxMoveDistance; y <= unitY + maxMoveDistance; y++) {
 					if(x < 0 || x >= grid.GetWidth() || y < 0 || y >= grid.GetHeight()) {
 						continue;
 					}
-					if(GameController.Instance.grid.GetGridObject(x, y).GetUnitGridCombat() != null) {
-						continue;
-					}
 
-					if(gridPathfinding.IsWalkable(x, y)) {
+					if(gridPathfinding.IsWalkable(x, y) && x != unitX || y != unitY) {

```

Comment:
What was unitX & unitY again? I assume current position?

Reply:
Those are the x and y coords of the selected unit.